create-branch/create-branch.sh

```diff
@@ -0,0 +1,76 @@
+#! /bin/bash
+set -eu
+
+echo "Create or checkout the branch."
+OWNER_REPO="${GITHUB_REPOSITORY}"
+git ls-remote --exit-code --heads https://github.com/${OWNER_REPO}.git refs/heads/$BRANCH || {
+  git branch $BRANCH $BASE_REF
+}
+git fetch origin $BRANCH || true
+git checkout $BRANCH
+
+echo "Get silk creds."
+# shellcheck disable=SC2046
+export $(grep -v '^#' $SILKBOMB_ENVFILE | xargs -0)
+
+echo "Get a silk token."
+SILK_JWT_TOKEN=$(curl -s -X POST "https://silkapi.us1.app.silk.security/api/v1/authenticate" \
+  -H "accept: application/json" -H "Content-Type: application/json" \
+  -d '{ "client_id": "'${SILK_CLIENT_ID}'", "client_secret": "'${SILK_CLIENT_SECRET}'" }' \
+  | jq -r '.token')
+
+echo "Get the silk asset group prefix."
+if [ -z "${SILK_PREFIX:-}" ]; then
+  REPO="${OWNER_REPO##*/}"
+  SILK_PREFIX=${REPO}
+fi
+SILK_GROUP="${SILK_PREFIX}-${BRANCH}"
+
+echo "Create the silk asset group."
+json_payload=$(cat <<EOF
+{
+    "active": true,
+    "name": "${SILK_GROUP}",
+    "code_repo_url": "https://github.com/${OWNER_REPO}",
+    "branch": "${BRANCH}",
+    "metadata": {
+        "sbom_lite_path": "${SBOM_FILE_PATH}"
+    },
+    "file_paths": [],
+    "asset_id": "$SILK_GROUP"
+}
+EOF
+)
+curl -X 'POST' \
+  'https://silkapi.us1.app.silk.security/api/v1/raw/asset_group' \
+  -H "accept: application/json" -H "Authorization: ${SILK_JWT_TOKEN}" \
+  -H 'Content-Type: application/json' \
+  -d "$json_payload"
+
+echo "SILK_ASSET_GROUP=$SILK_GROUP" >> $GITHUB_STEP_SUMMARY
+
+echo "Create a temp sbom."
+TMP_SBOM=sbom-for-${BRANCH}.json
+podman run --platform="linux/amd64" --rm -v "$(pwd)":/pwd \
+  ${ARTIFACTORY_IMAGE}/silkbomb:1.0 \
+  update --sbom-out /pwd/${TMP_SBOM}
+
+echo "Get the new timestamp and serial number."
+set -x
+SERIAL=$(jq -r '.serialNumber' ${TMP_SBOM})
+TIMESTAMP=$(jq -r '.metadata.timestamp' ${TMP_SBOM})
+rm ${TMP_SBOM}
+
+cat ${SBOM_FILE_PATH}
+echo "Replace the values in the existing sbom."
+cat <<< "$(jq --indent 4 '.serialNumber = "'${SERIAL}'"' ${SBOM_FILE_PATH})" > ${SBOM_FILE_PATH}
+cat <<< "$(jq --indent 4 '.metadata.timestamp = "'${TIMESTAMP}'"' ${SBOM_FILE_PATH})" > ${SBOM_FILE_PATH}
+cat ${SBOM_FILE_PATH}
+
+echo "Update the workflow with the silk asset group and evergreen project."
+sed -i 's/SILK_ASSET_GROUP:.*/SILK_ASSET_GROUP: '${SILK_GROUP}'/' ${RELEASE_WORKFLOW_PATH}
+sed -i 's/EVERGREEN_PROJECT:.*/EVERGREEN_PROJECT: '${EVERGREEN_PROJECT}'/' ${RELEASE_WORKFLOW_PATH}
```

Comment:
I think this relates to my earlier comment about the asset group prefix vs. hardcoded asset name. I'd personally change the release tooling to use the same prefix logic for the silk asset group.
As for evergreen project name, we don't use that in our PHP release workflow, but I can see how other drivers may need it. Wouldn't it make more sense to use the same `<prefix>-<branch-name>` for those names so we don't need to update this information for each release branch?

Reply:
We use the evergreen project name to tie the release to an Evergreen run url, which is included in the compliance report.  Doesn't each branch need its own Evergreen project?  Either way, we could make this part optional if no evergreen project is given.